collect_app/src/main/java/org/odk/collect/android/widgets/utilities/ExternalAppRecordingRequester.java

```diff
@@ -24,13 +27,24 @@
     private final ActivityAvailability activityAvailability;
     private final WaitingForDataRegistry waitingForDataRegistry;
     private final FormEntryViewModel formEntryViewModel;
+    private final AudioRecorderViewModel audioRecorderViewModel;
+    private final LifecycleOwner lifecycleOwner;
 
-    public ExternalAppRecordingRequester(Activity activity, ActivityAvailability activityAvailability, WaitingForDataRegistry waitingForDataRegistry, PermissionUtils permissionUtils, FormEntryViewModel formEntryViewModel) {
+    public ExternalAppRecordingRequester(Activity activity, ActivityAvailability activityAvailability, WaitingForDataRegistry waitingForDataRegistry, PermissionUtils permissionUtils, FormEntryViewModel formEntryViewModel, AudioRecorderViewModel audioRecorderViewModel, LifecycleOwner lifecycleOwner) {
         this.activity = activity;
         this.permissionUtils = permissionUtils;
         this.activityAvailability = activityAvailability;
         this.waitingForDataRegistry = waitingForDataRegistry;
         this.formEntryViewModel = formEntryViewModel;
+        this.audioRecorderViewModel = audioRecorderViewModel;
+        this.lifecycleOwner = lifecycleOwner;
+    }
+
+    @Override
+    public void onIsRecordingBlocked(Consumer<Boolean> isRecordingBlockedListener) {
+        audioRecorderViewModel.getCurrentSession().observe(lifecycleOwner, session -> {
```

Comment:
I did consider creating an abstract class that both `Internal...` and `External...` could extend from with this implementation. However, I feel like the code here **happens** to be the same as opposed to being the same. It could very well be that internal recordings are blocked for different reasons than external ones down the line.

Reply:
I'm not sure we're likely to add conditions but the principle makes sense to me and there isn't much more complexity one way or the other.